lib/samples/apply_style_to_wms_layer/apply_style_to_wms_layer.dart

```diff
@@ -0,0 +1,164 @@
+  const ApplyStyleToWmsLayer({super.key});
+
+  @override
+  State<ApplyStyleToWmsLayer> createState() => _ApplyStyleToWmsLayerState();
+}
+
+class _ApplyStyleToWmsLayerState extends State<ApplyStyleToWmsLayer>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  // Hold a reference to the layer to enable re-styling.
+  late WmsLayer _wmsLayer;
+
+  // String array to store the styles.
+  final _stylesTitles = [
+    'Default',
+    'Contrast stretch',
+  ];
+
+  // Create variable for holding sublayer style.
+  String? _selectedStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                // Build the bottom menu.
+                buildBottomMenu(),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: const SizedBox.expand(
+                child: ColoredBox(
+                  color: Colors.white30,
+                  child: Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildBottomMenu() {
+    return Center(
+      // A drop down button for selecting style.
+      child: DropdownButton(
+        alignment: Alignment.center,
+        hint: const Text(
+          'Choose a style',
+          style: TextStyle(
+            color: Colors.deepPurple,
+          ),
+        ),
+        icon: const Icon(Icons.arrow_drop_down),
+        iconEnabledColor: Colors.deepPurple,
+        iconDisabledColor: Colors.grey,
+        style: const TextStyle(color: Colors.deepPurple),
+        value: _selectedStyle,
+        items: _stylesTitles.map((items) {
+          return DropdownMenuItem(
+            value: items,
+            child: Text(items),
+          );
+        }).toList(),
+        onChanged: (style) {
+          if (style != null) {
+            changeStyle(style);
+          }
+        },
+      ),
+    );
+  }
+
+  void changeStyle(String style) async {
+    // Set the selected style.
+    setState(() => _selectedStyle = style);
+
+    // Get the available styles from the first sublayer.
+    final firstSublayer = _wmsLayer.sublayers.first as WmsSublayer;
+    final styles = firstSublayer.sublayerInfo.styles;
```

Comment:
I think we can follow my comment in the readme to instead get the styles from the `layerInfos` property on the `WmsLayer` etc.